File: app-android/app/src/main/java/com/health2u/domain/repository/UserRepository.kt
package com.health2u.domain.repository

import com.health2u.data.Result
import com.health2u.domain.model.User
import kotlinx.coroutines.flow.Flow

interface UserRepository {
    suspend fun getProfile(): Result<User>
    suspend fun updateProfile(user: User): Result<User>
    fun observeProfile(): Flow<User?>
    suspend fun logout(): Result<Unit>
}

File: app-ios/Sources/Health2u/Domain/Repository/UserRepository.swift
import Foundation

public protocol UserRepository: Sendable {
    func getProfile() async -> Result<User, APIError>
    func updateProfile(_ user: User) async -> Result<User, APIError>
    func observeProfile() -> AsyncStream<User?>
    func uploadProfilePhoto(imageData: Data, filename: String) async -> Result<User, APIError>
    func logout() async -> Result<Void, APIError>
}
